openhcl/virt_mshv_vtl/src/processor/snp/mod.rs

```diff
@@ -311,15 +363,16 @@ impl BackingPrivate for SnpBacked {
         this: &mut UhProcessor<'_, Self>,
         dev: &impl CpuIo,
         _stop: &mut virt::StopVp<'_>,
+        interrupt_pending: VtlArray<Option<u8>, 2>,
     ) -> Result<(), VpHaltReason<UhRunVpError>> {
-        this.run_vp_snp(dev).await
+        this.run_vp_snp(dev, interrupt_pending).await
     }
 
     fn poll_apic(
         this: &mut UhProcessor<'_, Self>,
         vtl: GuestVtl,
         scan_irr: bool,
-    ) -> Result<(), UhRunVpError> {
+    ) -> Result<Option<u8>, UhRunVpError> {
```

Comment:
The annoying bit there is keeping it backend-agnostic. The nice thing about this is we don't have to worry about SNP vs TDX, but if we try to read out of the VMSA or the TDX equivalent we would

Reply:
OK but this priority based thing is a mess, because you're redoing subtle work that we've already done.

Reply:
Fair. I guess I can inject a closure like i'm doing now for the lapic...

Reply:
I guess the problem is that for both SNP and TDX, you don't know if we've actually programmed the VMSA with an active interrupt; we just program the thing and let the hardware handle the PPR thing (but in two incompatible ways between SNP and TDX).

I still think you should just have a per-backing function that says "is there a pending interrupt that will be delivered", rather than try to recompute it generically.